File: bj/RobotCleaner_2_14503.py
N, M = map(int, input().split(' '))
r, c, d = map(int, input().split(' '))
rows = [list(map(int, input().split(' '))) for _ in range(N)]

di = [-1, 0, 1, 0]
dj = [0, 1, 0, -1]

ci = r
cj = c
cd = d
ans = 0

while True:
  if rows[ci][cj] == 0:
    rows[ci][cj] = 2
    ans += 1
  
  cantMove = True
  for k in range(4):
    ni = ci + di[k]
    nj = cj + dj[k]

    if not (ni >= 0 and ni < N and nj >= 0 and nj < M):
      continue

    if rows[ni][nj] == 0:
      cantMove = False
      break

  if cantMove:
    bd = (d + 2) % 4
    bi = ci + di[bd]
    bj = cj + dj[bd]

    if rows[bi][bj] == 1:
      break
    elif rows[bi][bj] == 2:
      ci = bi
      cj = bj
  else:
    d = (d + 3) % 4
    ni = ci + di[d]
    nj = cj + dj[d]

    if rows[ni][nj] == 0:
      ci = ni
      cj = nj

print(ans)

File: bj/RobotCleaner_2_14503.js
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

let inputs = [];
let N, M = undefined;
let r, c, d = undefined;
let count = 0, e = true;

readline.on('line', function(line) {
  if (N === undefined) {
    [N, M] = line.split(' ').map(el => parseInt(el));
    count = N;
  } else {
    if ( e ) {
      [r, c, d] = line.split(' ').map(el => parseInt(el));
      e = false;
      return; 
    }
    if (count>0) {
      inputs.push(line.split(' ').map(el => parseInt(el)))
      count --;
    } 
    if (count === 0) {
      readline.close();
    }
  }
}).on('close', function(){ //이 안에 솔루션 코드 작성
    solution(r, c, d, inputs);
    process.exit();
});

const solution = (r, c, d, rows) => {
  let ans = 0, conti = true, backIsWall = false;
  let dr = [-1, 0, 1, 0];
  let dc = [0, 1, 0, -1];
  
  while (conti) {
    if (rows[r][c] == 0) {
      ans ++;
      rows[r][c] = ans;

    }
    backIsWall = false;
    for (let i = 0; i < 5; i++) {
      let nd = (d + i) % 4;
      let nr = r + dr[nd];
      let nc = c + dc[nd];

      if (rows[nr][nc] == 0) {
        d = nd;
        r = nr;
        c = nc;
        break;
      }

      if (i == 2) {
        if (rows[nr][nc] == 1) {
          backIsWall = true;
        }
      }

      if (i == 4) {
        if (backIsWall) {
          conti = false;
          break;
        }
        else {
          nd = (d + 2) % 4;
          nr = r + dr[nd];
          nc = c + dc[nd];
          r = nr;
          c = nc;
          break;
        }
      }
    }
  }

  console.log('---')
  for (const row of rows) {
    console.log(row.join(' '))
  }
  console.log(ans);
};